core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java

```diff
@@ -84,11 +119,35 @@ public HasChildQueryBuilder setShortCircuitCutoff(int shortCircuitCutoff) {
     /**
      * Sets inner hit definition in the scope of this query and reusing the defined type and query.
      */
-    public HasChildQueryBuilder innerHit(QueryInnerHitBuilder innerHit) {
-        this.innerHit = innerHit;
+    public HasChildQueryBuilder innerHit(QueryInnerHits queryInnerHits) {
+        this.queryInnerHits = queryInnerHits;
         return this;
     }
 
+    public QueryBuilder getInnerQueryBuilder() {
```

Comment:
naming nitpick: all of the existing builders didn't move yet to get* getters, we planned on doing it later on. Does it make sense to use proper getters only for this object? I am leaning on keeping consistency with existing setters, remove the get prefix and change all of this at the same time when we are ready. Thoughts?